packages-node/providence-analytics/test-node/program/analyzers/match-imports.test.js

```diff
@@ -253,58 +253,206 @@ describe('Analyzer "match-imports"', () => {
+        },
+        {
+          exportSpecifier: {
+            filePath: './namespaced.js',
+            id: 'y::./namespaced.js::ref',
+            name: 'y',
+            project: 'ref',
+          },
+          matchesPerProject: [{ files: ['./index.js'], project: 'target' }],
+        },
+        {
+          exportSpecifier: {
+            filePath: './namespaced.js',
+            id: '[file]::./namespaced.js::ref',
+            name: '[file]',
+            project: 'ref',
+          },
+          matchesPerProject: [{ files: ['./index.js'], project: 'target' }],
+        },
+      ]);
+    });
+
+    describe('Inside small example project', () => {
+      it(`identifies all direct export specifiers consumed by "importing-target-project"`, async () => {
+        mockTargetAndReferenceProject(searchTargetProject, referenceProject);
+        await providence(matchImportsQueryConfig, _providenceCfg);
+        const queryResult = queryResults[0];
+        // console.log(JSON.stringify(queryResult.queryOutput, null, 2));
+        expectedExportIdsDirect.forEach(directId => {
+          expect(
+            queryResult.queryOutput.find(
+              exportMatchResult => exportMatchResult.exportSpecifier.id === directId,
+            ),
+          ).not.to.equal(undefined, `id '${directId}' not found`);
+        });
+      });
+
+      it(`identifies all indirect export specifiers consumed by "importing-target-project"`, async () => {
+        mockTargetAndReferenceProject(searchTargetProject, referenceProject);
+        await providence(matchImportsQueryConfig, _providenceCfg);
+        const queryResult = queryResults[0];
+        expectedExportIdsIndirect.forEach(indirectId => {
+          expect(
+            queryResult.queryOutput.find(
+              exportMatchResult => exportMatchResult.exportSpecifier.id === indirectId,
+            ),
+          ).not.to.equal(undefined, `id '${indirectId}' not found`);
+        });
+      });
+
+      it(`matches namespaced specifiers consumed by "importing-target-project"`, async () => {
+        mockTargetAndReferenceProject(searchTargetProject, referenceProject);
+        await providence(matchImportsQueryConfig, _providenceCfg);
+        const queryResult = queryResults[0];
+        expectedExportIdsNamespaced.forEach(exportedSpecifierId => {
+          expect(
+            queryResult.queryOutput.find(
+              exportMatchResult => exportMatchResult.exportSpecifier.id === exportedSpecifierId,
+            ),
+          ).not.to.equal(undefined, `id '${exportedSpecifierId}' not found`);
+        });
       });
     });
   });
 
   describe('Matching', () => {
     it(`produces a list of all matches, sorted by project`, async () => {
-      mockTargetAndReferenceProject(searchTargetProject, referenceProject);
-      await providence(matchImportsQueryConfig, _providenceCfg);
+      /**
+       * N.B. output structure could be simplified, since there is
```

Comment:
"Since there is ..."?

Reply:
Changed to:

```
* N.B. output structure could be simplified, since there is no need to order results by
* target project (there's only one target project per run).
* For now we keep it, so integration with dashboard stays intact.
```